tests/provider: SimpleDB domain test fails on GovCloud

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Description

This test fails on GovCloud:

```
TestAccAWSSimpleDBDomain_basic
```

With this error: `no such host`.

### New or Affected Resource(s)

<!--- Please list the new or affected resources and data sources. --->

* aws_simpledb_domain

### Terraform Configuration Files

Initial config:

```hcl
resource "aws_simpledb_domain" "test_domain" {
  name = "terraform-test-domain"
}
```

### Debug Output

```
    TestAccAWSSimpleDBDomain_basic: resource_aws_simpledb_domain_test.go:17: Step 1/2 error: Error running apply: 2020/11/05 17:41:24 [DEBUG] Using modified User-Agent: Terraform/0.12.29 HashiCorp-terraform-exec/0.10.0
        
        Error: Create SimpleDB Domain failed: RequestError: send request failed
        caused by: Post "https://sdb.us-gov-west-1.amazonaws.com/": dial tcp: lookup sdb.us-gov-west-1.amazonaws.com on 192.168.22.2:53: no such host

--- FAIL: TestAccAWSSimpleDBDomain_basic (28.24s)
```

### References

* #16062

--- a/aws/resource_aws_simpledb_domain_test.go
+++ b/aws/resource_aws_simpledb_domain_test.go
@@ -15,7 +15,7 @@ func TestAccAWSSimpleDBDomain_basic(t *testing.T) {
 	resourceName := "aws_simpledb_domain.test_domain"
 
 	resource.ParallelTest(t, resource.TestCase{
-		PreCheck:     func() { testAccPreCheck(t) },
+		PreCheck:     func() { testAccPreCheck(t); testAccPartitionHasServicePreCheck(simpledb.EndpointsID, t) },
 		Providers:    testAccProviders,
 		CheckDestroy: testAccCheckAWSSimpleDBDomainDestroy,
 		Steps: []resource.TestStep{

